lib/game.rb

```diff
@@ -0,0 +1,83 @@
+class Game
+
+  WIN_COMBINATIONS = [
+    [0, 1, 2], #top row
+    [3, 4, 5], #middle row
+    [6, 7, 8], #bottom row
+    [0, 3, 6], #first column
+    [1, 4, 7], #second column
+    [2, 5, 8], #third column
+    [0, 4, 8], #first diagonal
+    [2, 4, 6], #second diagonal
+  ]
+  
+  attr_accessor :board, :player_1, :player_2, :winner
+
+  def initialize(player_1=Human.new("X"), player_2=Human.new("O"), board=Board.new) 
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+    @winner = nil
+  end
+
+  def current_player
+    #if board#turn_count_moves is even, it's player_1. odd, it's player_2 
+    @board.turn_count % 2 == 0 ? player_1 : player_2
+  end
+
+  def over?
+    self.won? || self.draw?
+  end
+
+  def won?
+    WIN_COMBINATIONS.each do |combination|
```

Comment:
Here you can use [`Enumerable#detect`](http://ruby-doc.org/core-2.2.3/Enumerable.html#method-i-detect) to return the winning combination or `nil` if none is found.

```ruby
  def won?
    WIN_COMBINATIONS.find do |combo|
      @board.cells[combo[0]] == @board.cells[combo[1]] &&
      @board.cells[combo[0]] == @board.cells[combo[2]]
    end
  end
```